writer.go

```diff
@@ -177,81 +167,32 @@ func (re *WriterResponseEmitter) Emit(v interface{}) error {
 	return re.enc.Encode(v)
 }
 
-type Any struct {
-	types map[reflect.Type]bool
-	order []reflect.Type
+type MaybeError struct {
+	Value interface{} // needs to be a pointer
+	Error cmdkit.Error
```

Comment:
can we not assume that it's one or the other? (i.e., if Error is nil, it's a value)?

Reply:
hmm, `Error` is a struct so I can't check for `nil`. I could check if the message is empty but that seems like a bad hack. Checking whether `Value` is nil won't work because it's supposed to be set to a non-nil value by the user. So inferring whether we have an error or not is not possible in a clean way without `isError`.

I think the danger of incosistencies (i.e. `isError` is false but we in fact do have an error) is very small, so I'd like to stick with it.

Reply:
I see. Sounds good.